packages/tracing-internal/src/browser/browserTracingIntegration.ts

```diff
@@ -0,0 +1,545 @@
+    ? !!(
+        // eslint-disable-next-line deprecation/deprecation
+        (_options.tracePropagationTargets || _options.tracingOrigins)
+      )
+    : false;
+
+  addTracingExtensions();
+
+  // TODO (v8): remove this block after tracingOrigins is removed
+  // Set tracePropagationTargets to tracingOrigins if specified by the user
+  // In case both are specified, tracePropagationTargets takes precedence
+  // eslint-disable-next-line deprecation/deprecation
+  if (!_options.tracePropagationTargets && _options.tracingOrigins) {
+    // eslint-disable-next-line deprecation/deprecation
+    _options.tracePropagationTargets = _options.tracingOrigins;
+  }
+
+  const options = {
+    ...DEFAULT_BROWSER_TRACING_OPTIONS,
+    ..._options,
+  };
+
+  const _collectWebVitals = startTrackingWebVitals();
+
+  if (options.enableLongTask) {
+    startTrackingLongTasks();
+  }
+  if (options._experiments.enableInteractions) {
+    startTrackingInteractions();
+  }
+
+  let latestRouteName: string | undefined;
+  let latestRouteSource: TransactionSource | undefined;
+
+  /** Create routing idle transaction. */
+  function _createRouteTransaction(context: TransactionContext): Transaction | undefined {
+    // eslint-disable-next-line deprecation/deprecation
+    const hub = getCurrentHub();
+
+    const { beforeStartSpan, idleTimeout, finalTimeout, heartbeatInterval } = options;
+
+    const isPageloadTransaction = context.op === 'pageload';
+
+    const sentryTrace = isPageloadTransaction ? getMetaContent('sentry-trace') : '';
+    const baggage = isPageloadTransaction ? getMetaContent('baggage') : '';
+    const { traceparentData, dynamicSamplingContext, propagationContext } = tracingContextFromHeaders(
+      sentryTrace,
+      baggage,
+    );
+
+    const expandedContext: TransactionContext = {
+      ...context,
+      ...traceparentData,
+      metadata: {
+        // eslint-disable-next-line deprecation/deprecation
+        ...context.metadata,
+        dynamicSamplingContext: traceparentData && !dynamicSamplingContext ? {} : dynamicSamplingContext,
+      },
+      trimEnd: true,
+    };
+
+    const finalContext = beforeStartSpan ? beforeStartSpan(expandedContext) : expandedContext;
+
+    // If `beforeStartSpan` set a custom name, record that fact
+    // eslint-disable-next-line deprecation/deprecation
+    finalContext.metadata =
+      finalContext.name !== expandedContext.name
+        ? // eslint-disable-next-line deprecation/deprecation
+          { ...finalContext.metadata, source: 'custom' }
+        : // eslint-disable-next-line deprecation/deprecation
+          finalContext.metadata;
+
+    latestRouteName = finalContext.name;
+
+    // eslint-disable-next-line deprecation/deprecation
+    const sourceFromData = context.data && context.data[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE];
+    // eslint-disable-next-line deprecation/deprecation
+    const sourceFromMetadata = finalContext.metadata && finalContext.metadata.source;
```

Comment:
m: Shouldn't we additionally look for the source in `context.attributes[SEMANTIC_ATTRIBUTES_SENTRY_SOURCE]`?

Reply:
I just directly copied this over for now, `metadata` is a getter that merges that attribute in, so it should still work normally :) in v8 we adjust this then!

Reply:
ah wait nevermind, my comment is incorrect, I think you're right, let's also check attributes 👍